code/game/machinery/dehumidifier.dm

```diff
@@ -0,0 +1,103 @@
+//This magic dehumidifier is a portable water-to-nothing machine, although it does need a charged cell to work.
+//See drain.dm!
+/obj/machinery/dehumidifier
+	name = "emergency dehumidifier"
+	desc = "A portable dehumidifier with emergency high-volume pump to help keep you dry when it's getting damp... just very slowly."
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "dhum-off"
+	anchored = FALSE
+	density = TRUE
+
+	construct_state = /decl/machine_construction/default/panel_closed
+	uncreated_component_parts = null
+	base_type = /obj/machinery/dehumidifier
+	stat_immune = 0
+
+	power_channel = LOCAL
+	use_power = POWER_USE_OFF
+
+	atom_flags = ATOM_FLAG_CLIMBABLE
+	clicksound = "switch"
+
+	var/active = FALSE
+	var/drainage = TRUE
+	var/last_gurgle = 0
+
+/obj/machinery/dehumidifier/Initialize()
+	. = ..()
+	queue_icon_update()
+
+/obj/machinery/dehumidifier/on_update_icon(rebuild_overlay = FALSE)
```

Comment:
Nothing will ever call this with `TRUE` there, so it will never get the overlay. Just make it add it always; not like this is expensive.